scripts-dev/generate_full_schema.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+#
+# This script generates SQL files for creating a brand new Synapse DB with the latest
+# schema, on both SQLite3 and Postgres.
+#
+# It does so by having Synapse generate an up-to-date SQLite DB, then running
+# synapse_port_db to convert it to Postgres. It then dumps the contents of both.
+
+POSTGRES_HOST="localhost"
+POSTGRES_DB_NAME="synapse_full_schema"
+SQLITE_DB_FILE_NAME="full_schema_sqlite.db"
+
+SQLITE_FULL_SCHEMA_OUTPUT_FILE="full.sql.sqlite"
+POSTGRES_FULL_SCHEMA_OUTPUT_FILE="full.sql.postgres"
+OUTPUT_DIR=$(pwd)
+
+usage() {
+  echo "Usage: $0 -p <postgres_username> [-c] [-v] [-o] [-h]"
+  echo
+  echo "-p <postgres_username>"
+  echo "  Username to connect to local postgres instance. The password will be requested"
+  echo "  during script execution."
+  echo "-c"
+  echo "  Whether to enable coverage tracking. Useful for CI runs."
+  echo "-v"
+  echo "  Suppress warning about requiring the use of a virtualenv."
+  echo "-o"
+  echo "  Directory to output full schema files to. Defaults to the current directory."
+  echo "-h"
+  echo "  Display this help text."
+}
+
+while getopts "p:cvo:h" opt; do
+  case $opt in
+    p)
+      POSTGRES_USERNAME=$OPTARG
+      ;;
+    c)
+      COVERAGE=1
+      ;;
+    v)
+      NO_VIRTUALENV=1
+      ;;
+    o)
+      command -v realpath > /dev/null || (echo "The -o flag requires the 'realpath' binary to be installed" && exit 1)
+      OUTPUT_DIR="$(realpath "$OPTARG")"
+      ;;
+    h)
+      usage
+      exit
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit
+      ;;
+  esac
+done
+
+# Check that the script is running with a virtualenv enabled
+if [ -z ${NO_VIRTUALENV+x} ] && [ -z ${VIRTUAL_ENV+x} ]; then
+  echo "It is highly recommended to run this script with a virtualenv activated. Exiting now."
+  echo "If you wish to suppress this warning, please run with the -v option."
+  exit 1
+fi
+
+if [ -z ${POSTGRES_USERNAME+x} ]; then
+  echo "No postgres username supplied"
+  usage
+  exit 1
+fi
+
+read -rsp "Postgres password for '$POSTGRES_USERNAME':" POSTGRES_PASSWORD
+echo ""
+
+set -xe
+
+# cd to root of the synapse directory
+cd "$(dirname "$0")/.."
+
+# Install required dependencies
+echo "Installing dependencies..."
+if [ -z ${COVERAGE+x} ]; then
+  # No coverage needed
+  pip install psycopg2
+else
+  # Coverage desired
+  pip install psycopg2 coverage coverage-enable-subprocess
+fi
+
+# Install Synapse itself. This won't update any libraries.
+pip install -e .
```

Comment:
Ah, this was the reason for prompting the user to run with a virtualenv enabled, but I agree that it's not obvious that it will be modified.

I've changed the script now so it prints out which packages you need to install before it runs.